modules/nf-core/sourmash/sketch/main.nf

```diff
@@ -18,7 +18,7 @@ process SOURMASH_SKETCH {
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: "dna --param-string 'scaled=1000,k=31'"
+    def args = task.ext.args ?: "dna --param-string 'scaled=1000,k=21,k=31,k=51,abund'"
```

Comment:
Please note that this should be the absolute default arguments for this kind of process. Any more specialized application is expected to override `task.ext.args`. Are you sure these should be the default ones?

Reply:
yep.

(Except for `abund`, these are the default arguments inside of sourmash for `sketch dna` as well, but I think it's good to have them explicit here.)

Reply:
I checked in with other sourmash devs @ctb and @bluegenes and they agree that the default should be `scaled=1000,k=21,k=31,k=51,abund`. The idea is that signatures should be a unit that is kept together in a single file, and that you should only read the data once to get all of the sketches.

Reply:
oh woops sorry i didn't reload the page before I posted this to see that titus had responded! 😂

Reply:
Isn't it a bit weird that you should manually define the arguments that are seen as default? (just curious why it is this way :p)

Reply:
Okay, just wanted to make sure. Thanks for the response @taylorreiter and @ctb 

@nvnieuwk that depends on your definition of manual. These modules are expected to be included in a variety of pipelines and it is in those pipeline configurations where the expected arguments should be set for the particular application of this module within the pipeline.

Reply:
Aaah okay that makes sense! Thanks for clarifying!

Reply:
I had the same  question, @taylorreiter could you add a comment just above the line specify this (otherwise this might come up again, embedding `args` is very unusual/normally frowned upon).

Reply:
(not familiar with nf-core policies, so take this as you will ;)

on reflection, since sourmash uses semantic versioning, `scaled=1000,k=21,k=31,k=51` are default and only `-p abund` is not, you might want to omit the first part - as long as this workflow is pinned to sourmash<5, you'll be good.

on the other hand it could be a nice matter of user education since these parameters are among the most likely ones to be tweaked by the user.